java/README.md

```diff
@@ -49,17 +49,35 @@ In the `tasks.withType(JavaCompile)` section, we pass some configuration options
 The properties file has the following template: 
 
 ```
-treatedMethods=treated,flagEnabled
-controlMethods=flagDisabled
-emptyMethods=enableFlag,disableFlag
-treatmentGroupMethods=isToggleInGroup
-annotations=FlagTesting
-linkURL=<provide_your_url>
+{
+  "methodProperties":
+    [
+      {
+        "methodName": "isToggleEnabled",
+        "flagType": "treated",
+        "returnType": "boolean",
+        "receiverType": "com.uber.piranha.XPTest",
+        "argumentIndex": 0
+      },
+      ...
+    ],
+  "linkURL": "<provide_your_url>",
+  "annotations": ["ToggleTesting"]
+}
 ```
 
-The `treatedMethods` are the APIs which correspond to the treatment behavior of the flag, `controlMethods` correspond to the control behavior of the flag. In the above example, the API `flagEnabled` corresponds to treatment behavior. Hence, when `IsTreated` flag is set to `true`, `flagEnabled(SAMPLE_STALE_FLAG)` will be evaluated to `true`. Similarly, `flagDisabled(SAMPLE_STALE_FLAG)` which corresponds to the control behavior will evaluate to `false`. 
+The required top-level field is `methodProperties`.
+Within that, there is an array of JSON objects, having the required fields `methodName` and `flagType`.
+The optional fields are `returnType`, `receiverType` and `argumentIndex`.
+
+The `flagType` with `treated` are the APIs which correspond to the treatment behavior of the flag, `control` correspond to the control behavior of the flag. In the above example, the API `flagEnabled` corresponds to treatment behavior. Hence, when `IsTreated` Piranha argument is set to `true`, `flagEnabled(SAMPLE_STALE_FLAG)` will be evaluated to `true`. Similarly, `flagDisabled(SAMPLE_STALE_FLAG)` which corresponds to the control behavior will evaluate to `false`. 
+
+The `flagType` with `empty` specifies the APIs which need to be discarded from the code. For example, a statement `enableFlag(SAMPLE_STALE_FLAG);` will be deleted from the code. 
+
+The optional field `argumentIndex` specifies where to look for the flag name (given by `-XepOpt:Piranha:FlagName`) in the method's arguments. We follow 0 based indexing.
+If your toggle method takes no arguments, or if you want to delete all occurrences of a given `methodName` irrespective of their arguments, `argumentIndex` need not be specified.
 
-The `emptyMethods` specify the APIs which need to be discarded from the code. For example, a statement `enableFlag(SAMPLE_STALE_FLAG);` will be deleted from the code. 
+For `returnType` and `receiverType`, types should be written as `boolean` or `void` for inbuilt types, and fully qualified for custom defined types. eg: `com.uber.piranha.XPFlagCleanerPositiveCases.XPTest`
```

Comment:
inbuilt -> primitive.

Reply:
Also added java.lang.string and a java.lang.boolean (which returns `Boolean` instead of `boolean`) test